tools/setup_perf_cluster.sh

```diff
@@ -33,17 +33,41 @@ function Usage() {
     exit 1
 }
 
+function abspath() {
+# Source https://stackoverflow.com/questions/3915040/bash-fish-command-to-print-absolute-path-to-a-file
+# Thanks to Alexander Klimetschek
+
+    # generate absolute path from relative path
+    # $1     : relative filename
+    # return : absolute path
+    if [ -d "$1" ]; then
+        # dir
+        (cd "$1"; pwd)
+    elif [ -f "$1" ]; then
+        # file
+        if [[ $1 = /* ]]; then
+            echo "$1"
+        elif [[ $1 == */* ]]; then
+            echo "$(cd "${1%/*}"; pwd)/${1##*/}"
+        else
+            echo "$(pwd)/$1"
+        fi
+    fi
+}
+
 function List_functions() {
   grep -E "^function [a-z]" "${BASH_SOURCE[0]}" | sed -e 's/function \([a-z_0-9]*\).*/\1/'
 }
 
 if [[ "${BASH_SOURCE[0]}" != "${0}" ]]; then
-  TOOLS_DIR=${TOOLS_DIR:-$(dirname "${BASH_SOURCE[0]}")}
+  TOOLS_ABSPATH=$(abspath ${BASH_SOURCE[0]})
```

Comment:
what is this for? I sort of remember abspath isn’t everywhere and the red version works on both mac and linux

Reply:
`abspath` convert relative path to absolute path to make sure that the scripts and templates can be found.